Kitodo-MODSXML-SchemaConverter/src/main/java/org/kitodo/modsxmlschemaconverter/ModsXMLSchemaConverter.java

```diff
@@ -0,0 +1,138 @@
+import org.kitodo.api.schemaconverter.DataRecord;
+import org.kitodo.api.schemaconverter.FileFormat;
+import org.kitodo.api.schemaconverter.MetadataFormat;
+import org.kitodo.api.schemaconverter.SchemaConverterInterface;
+import org.kitodo.exceptions.ConfigException;
+import org.xml.sax.InputSource;
+
+public class ModsXMLSchemaConverter implements SchemaConverterInterface {
+
+    private static final Logger logger = LogManager.getLogger(ModsXMLSchemaConverter.class);
+
+    private static final String XSLT_FILEPATH = "/xslt/mods2kitodo.xsl";
+
+    private static MetadataFormat supportedSourceMetadataFormat = MetadataFormat.MODS;
+    private static MetadataFormat supportedTargetMetadataFormat = MetadataFormat.KITODO;
+    private static FileFormat supportedSourceFileFormat = FileFormat.XML;
+    private static FileFormat supportedTargetFileFormat = FileFormat.XML;
+
+    /**
+     * Converts a given DataRecord to the given MetadataFormat 'targetMetadataFormat' and FileFormat 'targetFileFormat'.
+     *
+     * @param record DataRecord to be converted
+     * @param targetMetadataFormat MetadataFormat to which the given DataRecord is converted
+     * @param targetFileFormat FileFormat to which the given DataRecord is converted
+     * @return The result of the conversion as a DataRecord.
+     */
+    @Override
+    public DataRecord convert(DataRecord record, MetadataFormat targetMetadataFormat, FileFormat targetFileFormat) {
+        if (!(supportsSourceMetadataFormat(record.getMetadataFormat())
+                && supportsSourceFileFormat(record.getFileFormat())
+                && supportsTargetMetadataFormat(targetMetadataFormat)
+                && supportsTargetFileFormat(targetFileFormat))) {
+            throw new UnknownFormatConversionException("Unable to convert from " + record.getMetadataFormat() + "/"
+                    + record.getFileFormat() + " to " + targetMetadataFormat + "/" + targetFileFormat + "!");
+        }
+
+        if (record.getOriginalData() instanceof InputStream) {
+            try (InputStream inputStream = (InputStream)record.getOriginalData();
+                 InputStream fileStream = getClass().getResourceAsStream(XSLT_FILEPATH)) {
+                if (Objects.isNull(fileStream)) {
+                    logger.error("Unable to load XSL transformation file!");
+                    throw new IOException("Unable to load XSL transformation file!");
+                }
+                String xmlString = IOUtils.toString(inputStream, Charset.defaultCharset());
+                DataRecord resultRecord = new DataRecord();
+                resultRecord.setOriginalData(transformXmlByXslt(xmlString, fileStream));
+                resultRecord.setFileFormat(targetFileFormat);
+                resultRecord.setMetadataFormat(targetMetadataFormat);
+                return resultRecord;
+            } catch (IOException e) {
+                logger.error(e.getMessage());
+            }
+        }
+        return null;
```

Comment:
I'm not a fan of returning `null` values. Is there no other way to avoid this? Inside the if condition before you are catching all `IOException` and logging this. After this the method is returning `null`. I don't think this is a good way / behaviour. Last remark to this: In the JAvaDoc of this method is no information that this method returns a `null` value.